pinot-broker/src/main/java/org/apache/pinot/broker/broker/helix/BaseBrokerStarter.java

```diff
@@ -439,17 +440,26 @@ private void updateInstanceConfigAndBrokerResourceIfNeeded() {
     }
     updated |= HelixHelper.removeDisabledPartitions(instanceConfig);
     boolean shouldUpdateBrokerResource = false;
-    String brokerTag = null;
     List<String> instanceTags = instanceConfig.getTags();
     if (instanceTags.isEmpty()) {
       // This is a new broker (first time joining the cluster)
       if (ZKMetadataProvider.getClusterTenantIsolationEnabled(_propertyStore)) {
-        brokerTag = TagNameUtils.getBrokerTagForTenant(null);
+        instanceConfig.addTag(TagNameUtils.getBrokerTagForTenant(null));
         shouldUpdateBrokerResource = true;
       } else {
-        brokerTag = Helix.UNTAGGED_BROKER_INSTANCE;
+        String instanceTagsConfig = _brokerConf.getProperty(Broker.CONFIG_OF_BROKER_INSTANCE_TAGS);
```

Comment:
@Jackie-Jiang why is this set only if cluster tenant isolation is false? i was just testing this and noticed it wasn't working on our cluster, but also noticed cluster isolation defaults to true.

Reply:
@dang-stripe The method name could be confusing, but when it is enabled, it means single tenant cluster, where all instances should be under the same tenant. That is why we ignore the config.